components/persistor/lib/knex/update.ts

```diff
@@ -140,8 +140,12 @@ module.exports = function (PersistObjectTemplate) {
                     value.setDirty(txn);
                 }
 
-                pojo[foreignKey] =  value ? value._id : null
-                updatePersistorProp(obj, prop + 'Persistor', {isFetching: false, id: value ? value._id : null, isFetched: true})
+                pojo[foreignKey] = value ? value._id : null;
```

Comment:
You can probably use an optional chain if you want here. Also can we use this variable in the block below instead of constructing it again?

```suggestion
                pojo[foreignKey] = value?._id;
```